### PR TITLE
Initialize generated tensor data in onnxruntime_perf_test.

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**
Initialize generated tensor data in onnxruntime_perf_test to zeroes instead of leaving it uninitialized. String tensors were already being initialized.

**Motivation and Context**
Avoid reading uninitialized data and getting inconsistent input values.

### CODE REVIEW

File: onnxruntime/test/perftest/ort_test_session.cc [hunk 1]
```diff
@@ -563,6 +564,47 @@ select from 'TF8', 'TF16', 'UINT8', 'FLOAT', 'ITENSOR'. \n)");
   }
 }
 
+template <typename T>
+static void FillTensorDataTyped(Ort::Value& tensor, size_t count, T value = T{}) {
+  T* data = tensor.GetTensorMutableData<T>();
+  std::fill_n(data, count, value);
+}
```
Comment (yuslepukhin) [inline onnxruntime/test/perftest/ort_test_session.cc:571]: I would say simply do memset(tensor.MutableDataRaw(), 0, tensor.SizeInBytes()) and no switch or templates.
Another idea is to emulate other frameworks and add SetToZero() method to Tensor class?

Comment (edgchen1) [reply onnxruntime/test/perftest/ort_test_session.cc:571]: is there an equivalent to `tensor.SizeInBytes()` in the C++ API?

Comment (yuslepukhin) [reply onnxruntime/test/perftest/ort_test_session.cc:571]: No. Our API needs work.

Comment (edgchen1) [reply onnxruntime/test/perftest/ort_test_session.cc:571]: looks like this file is using the C++ API and not the internal classes. was hoping to keep this PR's fix local to onnxruntime_perf_test and not have to add new public APIs

Comment (yuslepukhin) [reply onnxruntime/test/perftest/ort_test_session.cc:571]: ok

**Review by yuslepukhin — APPROVED** [review]
:shipit:

**Pull request merged**